<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>784</width>
    <height>440</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cherygova-cg16 Lab3</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>562</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="1">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>103</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="1">
    <widget class="QTabWidget" name="Tab">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="GeneralSettings">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="1">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Figure:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Light model:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1" colspan="2">
        <widget class="QComboBox" name="Figure">
         <item>
          <property name="text">
           <string>Torus</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Sphere</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="3" column="1" colspan="2">
        <widget class="QComboBox" name="LightModel">
         <item>
          <property name="text">
           <string>Blinn-Fong</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Cook-Torrance</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QSpinBox" name="StepsSpinBox">
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>2</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QSlider" name="StepsSlider">
         <property name="minimum">
          <number>2</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Steps:</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="LightSourceSettings">
      <attribute name="title">
       <string>Light</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="1">
        <widget class="QLabel" name="label_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignBottom|Qt::AlignHCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QSpinBox" name="LSourceSpinX">
         <property name="acceptDrops">
          <bool>true</bool>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>-1000</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QScrollBar" name="LSourceScrollZ">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-1000</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QScrollBar" name="LSourceScrollY">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-1000</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="label_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Z</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignBottom|Qt::AlignHCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>X</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignBottom|Qt::AlignHCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="LSourceSpinY">
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>-1000</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="LSourceSpinZ">
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>-1000</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QScrollBar" name="LSourceScrollX">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-1000</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="ColorSettings">
      <attribute name="title">
       <string>Color</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="6" column="1">
        <widget class="QSlider" name="GreenSlider3">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="9" column="1">
        <widget class="QSlider" name="BlueSlider3">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <widget class="QSlider" name="BlueSlider2">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QSlider" name="GreenSlider2">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QSlider" name="BlueSlider1">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSlider" name="GreenSlider1">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="5" rowspan="9">
        <widget class="QSlider" name="CoeffSlider">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSlider" name="RedSlider2">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSlider" name="RedSlider1">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSlider" name="RedSlider3">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="GreenLabel">
         <property name="text">
          <string>green</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="RedLabel">
         <property name="text">
          <string>red</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="BlueLabel">
         <property name="text">
          <string>blue</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="4" colspan="2">
        <widget class="QLabel" name="CoeffLabel">
         <property name="text">
          <string>alpha / m</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>Tab</tabstop>
  <tabstop>Figure</tabstop>
  <tabstop>LightModel</tabstop>
  <tabstop>StepsSpinBox</tabstop>
  <tabstop>StepsSlider</tabstop>
  <tabstop>LSourceSpinX</tabstop>
  <tabstop>LSourceSpinY</tabstop>
  <tabstop>LSourceSpinZ</tabstop>
  <tabstop>RedSlider1</tabstop>
  <tabstop>RedSlider2</tabstop>
  <tabstop>RedSlider3</tabstop>
  <tabstop>GreenSlider1</tabstop>
  <tabstop>GreenSlider2</tabstop>
  <tabstop>GreenSlider3</tabstop>
  <tabstop>BlueSlider1</tabstop>
  <tabstop>BlueSlider2</tabstop>
  <tabstop>BlueSlider3</tabstop>
  <tabstop>CoeffSlider</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>LSourceScrollY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LSourceSpinY</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>540</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LSourceScrollX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LSourceSpinX</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>607</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>610</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LSourceScrollZ</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LSourceSpinZ</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>727</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>730</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LSourceSpinX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LSourceScrollX</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>610</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>607</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LSourceSpinY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LSourceScrollY</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>670</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>667</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LSourceSpinZ</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LSourceScrollZ</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>730</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>727</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StepsSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>StepsSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>642</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>509</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StepsSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>StepsSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>509</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>642</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
